servers/rendering/renderer_scene_render.cpp

```diff
@@ -33,17 +33,18 @@
 /////////////////////////////////////////////////////////////////////////////
 // CameraData
 
-void RendererSceneRender::CameraData::set_camera(const Transform3D p_transform, const Projection p_projection, bool p_is_orthogonal, bool p_vaspect, const Vector2 &p_taa_jitter, const uint32_t p_visible_layers) {
+void RendererSceneRender::CameraData::set_camera(const Transform3D p_transform, const Projection p_main_projection, const Projection p_shadow_projection, bool p_is_orthogonal, bool p_vaspect, const Vector2 &p_taa_jitter, const uint32_t p_visible_layers) {
 	view_count = 1;
 	is_orthogonal = p_is_orthogonal;
 	vaspect = p_vaspect;
 
 	main_transform = p_transform;
-	main_projection = p_projection;
+	main_projection = p_main_projection;
+	shadow_projection = p_shadow_projection;
```

Comment:
I'd rather call it `culling_projection` as it is what it is used for in your code (except the open question of effects).